### PR TITLE
status: add dnsrecords to related objects

### PR DESCRIPTION
**Pull request by ironcladlou**
Add dnsrecords to related objects to ensure they are collected by must-gather.

### CODE REVIEW

File: pkg/operator/controller/status/controller.go [hunk 1]
```diff
@@ -168,23 +179,41 @@ func initializeClusterOperator(co *configv1.ClusterOperator) {
...
+type operatorState struct {
+	Namespace          *corev1.Namespace
+	IngressControllers []operatorv1.IngressController
+	DNSRecords         []iov1.DNSRecord
+}
+
 // getOperatorState gets and returns the resources necessary to compute the
 // operator's current state.
-func (r *reconciler) getOperatorState(nsName string) ([]operatorv1.IngressController, *corev1.Namespace, error) {
+func (r *reconciler) getOperatorState(nsName string) (*operatorState, error) {
```
Comment (Miciah) [inline pkg/operator/controller/status/controller.go]: Is there any reason `operatorState` should be a pointer value?

Comment (ironcladlou) [reply pkg/operator/controller/status/controller.go]: No good reason — changed to a value type

File: pkg/operator/controller/status/controller.go [hunk 2]
```diff
@@ -168,23 +179,41 @@ func initializeClusterOperator(co *configv1.ClusterOperator) {
...
 	ingressList := &operatorv1.IngressControllerList{}
 	if err := r.cache.List(context.TODO(), ingressList, client.InNamespace(r.Namespace)); err != nil {
-		return nil, nil, fmt.Errorf("failed to list IngressControllers: %v", err)
+		return nil, fmt.Errorf("failed to list ingresscontrollers in %q: %v", r.Namespace, err)
+	} else {
+		state.IngressControllers = ingressList.Items
+	}
+
+	dnsRecords := &iov1.DNSRecordList{}
+	if err := r.cache.List(context.TODO(), ingressList, client.InNamespace(r.Namespace)); err != nil {
```
Comment (Miciah) [inline pkg/operator/controller/status/controller.go]: Bad copy pasta: `ingressList`→`dnsRecords`.

Comment (ironcladlou) [reply pkg/operator/controller/status/controller.go]: Hopefully the tests failed for that... thanks

**Comment by Miciah** [comment]
[The clusteroperator references the `default-wildcard` DNSRecord](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-ingress-operator/281/pull-ci-openshift-cluster-ingress-operator-master-e2e-aws/1292/artifacts/e2e-aws/must-gather/cluster-scoped-resources/config.openshift.io/clusteroperators/ingress.yaml), and [the DNSRecord was gathered](https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-ingress-operator/281/pull-ci-openshift-cluster-ingress-operator-master-e2e-aws/1292/artifacts/e2e-aws/must-gather/namespaces/openshift-ingress-operator/ingress.operator.openshift.io/dnsrecords/default-wildcard.yaml).

/lgtm

**Pull request merged**